compiler/qsc_rir/src/passes/reindex_qubits/tests.rs

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![allow(clippy::too_many_lines, clippy::needless_raw_string_hashes)]
+
+use expect_test::expect;
+
+use crate::{
+    builder::{cx_decl, h_decl, mresetz_decl, mz_decl, reset_decl, x_decl},
+    rir::{Block, BlockId, CallableId, CallableType, Instruction, Literal, Program, Value},
+};
+
+use super::reindex_qubits;
+
+#[test]
+fn qubit_reindexed_after_reset_removes_reset() {
+    let mut program = Program::new();
+    program.num_qubits = 1;
+    program.callables.insert(CallableId(0), x_decl());
+    program.callables.insert(CallableId(1), reset_decl());
```

Comment:
One thing that would be helpful when reading these tests would be to use constants for callable IDs:
 ```rust
const X: usize = 0;
const RESET: usize = 1;
program.callables.insert(X.into(), x_decl());
program.callables.insert(RESET.into(), reset_decl());
 ```
 
 This way a sequence of call instructions would be easier to read:
 ```rust
Instruction::Call(X.into(), vec![Value::Literal(Literal::Qubit(0))], None),
Instruction::Call(RESET.into(), vec![Value::Literal(Literal::Qubit(0))], None),
 ```

Reply:
That's a good idea, thanks!